projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * @license
+ * Alfresco Example Content Application
+ *
+ * Copyright (C) 2005 - 2020 Alfresco Software Limited
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { AppConfigService } from '@alfresco/adf-core';
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { OpenInAppComponent } from '../components/open-in-app/open-in-app.component';
+
+export interface MobileAppSwitchConfigurationOptions {
+  enabled: string;
+  isIphone: string;
+  isAndroidPart1: string;
+  isAndroidPart2: string;
+  sessionTimeForOpenAppDialogDisplay: string;
+}
+@Injectable({
+  providedIn: 'root'
+})
+export class AcaMobileAppSwitcherService {
+  private mobileAppSwitchConfig: MobileAppSwitchConfigurationOptions;
+  public redirectUrl: string;
+
+  constructor(private config: AppConfigService, private dialog: MatDialog) {
+    this.mobileAppSwitchConfig = this.config.get<MobileAppSwitchConfigurationOptions>('mobileAppSwitch');
+  }
+
+  checkForMobileApp(): void {
+    const currentTime: number = new Date().getTime();
+    const sessionTime: string = sessionStorage.getItem('sessionTime');
+    const sessionConvertedTime: number = parseFloat(sessionTime);
```

Comment:
you don't even check if session time was previously defined here! on the new machine the parseFloat will give you "NaN" that you will then be trying to substract from the current time? this needs review and unit testing

Reply:
I have applied check for sessionTime now if it has some value then only the conversion happens otherwise not.